xapi/futures/xenopsd_events.md

```diff
@@ -0,0 +1,47 @@
+---
+layout: default
+title: Process events from xenopsd in a timely manner
+design_doc: true
+status: proposed
+revision: 1
+---
+
+# Background
+
+There is a significant delay between the VM being unpaused and XAPI reporting it
+as started during a bootstorm.
+It can happen that the VM is able to send UDP packets already, but XAPI still reports it as not started for minutes.
+
+XAPI currently processes all events from xenopsd in a single thread, the unpause
```

Comment:
I assume the relevant code that processes events from xenopsd in xapi is somewhere here: https://github.com/xapi-project/xen-api/blob/master/ocaml/xapi/xapi_xenops.ml#L1204